README.md

```diff
@@ -4,24 +4,31 @@
 Small wrapper to ease creating services that read from Azure Storage Queues
 
 ## Usage
-Create a new Worker service project, and install this package. Then change the base class of your `Worker` to `AzureStorageQueueWorker<T>` where `T` is the type of message you expect to receive (a POCO object):
+Create a new Worker service project, and install this package.
+Implement `IMessageProcessor<TMessage>` where `TMessage` is the type of message you expect to receive (a POCO object, string or QueueMessage):
 
 ```
-public class Worker : AzureStorageQueueWorker<MyCustomMessage>
+public class MessageProcessor : IMessageProcessor<MyCustomMessage>
 {
     private readonly ILogger<Worker> logger;
 
-    public Worker(
-        ILogger<Worker> logger,
-        IConfiguration configuration)
-        : base(logger, configuration["AZURE_STORAGE_CONNECTION_STRING"], configuration["QUEUE_NAME"])
+    public Worker(ILogger<Worker> logger)
     {
         this.logger = logger;
     }
 
-    protected override async Task ProcessMessage(MyCustomMessage message, CancellationToken stoppingToken)
+    public async Task ProcessMessage(MyCustomMessage message, CancellationToken stoppingToken)
     {
         // TODO: Process the message
     }
 }
 ```
+Register created class in Program:
+```
+services.AddMessageProcessor<MyCustomMessage, MessageProcessor>(
+    (sp, b) =>
+    {
+        b.Options.AzureStorageConnectionString = "";
+        b.Options.QueueName = "";
+    });
```

Comment:
Maybe also add a note about enabling application insights telemetry, now that we have taken a dependency on it:

`services.AddApplicationInsightsTelemetryWorkerService();`

Reply:
but default it'll be enabled. I'll add a note that it can be disabled via param but then it should be configured in the consumption code

Reply:
Ahh, sorry - I missed that you enable it inside `AzureStorageQueueWorkerServiceCollectionExtensions` 👍